### PR TITLE
10216: rework PublicCase.isPrivateDocument and accompanying tests

### PR DESCRIPTION
**Pull request by mmarcotte**

### CODE REVIEW

File: shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts [hunk 1]
```diff
@@ -1,155 +1,343 @@
...
+                  },
+                },
+                visibilityChangeDate,
+              );
+
+              expect(isPrivate).toEqual(false);
+            },
+          );
+
+          // there's currently no good way to determine this information
```
Comment (ujahio) [inline shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts:220]: ❓ : Are these test scenarios going to be addressed or can they be deleted?

File: shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts [hunk 2]
```diff
@@ -1,155 +1,343 @@
...
+                  docketEntryId: '123',
+                  eventCode,
+                  filedByRole,
+                  filingDate,
+                  isOnDocketRecord: true,
+                  isPaper: false,
+                  previousDocument: {
+                    docketEntryId: '2a389787-91d2-41ba-9c6e-2a13440a928b',
+                    documentType: 'Seriatim Answering Brief',
+                    // filedByRole: 'privatePractitioner'
```
Comment (ujahio) [inline shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts]: ❓ :  Is `filedByRole` still needed as a prop in previousDocument?

File: shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts (hunk 1 above)
Comment (mmarcotte) [reply shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts:220]: they are going to have to be addressed, but not in the scope of this pr

File: shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts (hunk 2 above)
Comment (mmarcotte) [reply shared/src/business/entities/cases/PublicCase.isPrivateDocument.test.ts]: it is still needed, but it is not stored that way.

**Review by ujahio — APPROVED** [review]

**Pull request merged**